src/components/MobileWalletDialog/routes/SavedWallets.tsx

```diff
@@ -0,0 +1,124 @@
+import { ChatIcon, SettingsIcon } from '@chakra-ui/icons'
+import { Button, Stack } from '@chakra-ui/react'
+import { WalletConnectToDappsHeaderButton } from 'plugins/walletConnectToDapps/components/header/WalletConnectToDappsHeaderButton'
+import { useCallback, useMemo, useState } from 'react'
+import { TbCircleArrowDown, TbCirclePlus } from 'react-icons/tb'
+import { useTranslate } from 'react-polyglot'
+import { MainNavLink } from 'components/Layout/Header/NavBar/MainNavLink'
+import { DialogBody } from 'components/Modal/components/DialogBody'
+import { DialogFooter } from 'components/Modal/components/DialogFooter'
+import {
+  DialogHeader,
+  DialogHeaderMiddle,
+  DialogHeaderRight,
+} from 'components/Modal/components/DialogHeader'
+import { DialogTitle } from 'components/Modal/components/DialogTitle'
+import { SlideTransition } from 'components/SlideTransition'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { KeyManager } from 'context/WalletProvider/KeyManager'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { useModal } from 'hooks/useModal/useModal'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { MobileWalletList } from 'pages/ConnectWallet/components/WalletList'
+
+const addIcon = <TbCirclePlus />
+const importIcon = <TbCircleArrowDown />
+const settingsIcon = <SettingsIcon />
+const chatIcon = <ChatIcon />
+
+type SavedWalletsProps = {
+  onClose: () => void
+}
+
+export const SavedWallets: React.FC<SavedWalletsProps> = ({ onClose }) => {
+  const translate = useTranslate()
+  const settings = useModal('settings')
+  const feedbackSupport = useModal('feedbackSupport')
+  const isWalletConnectToDappsV2Enabled = useFeatureFlag('WalletConnectToDappsV2')
+  const { dispatch, create, importWallet } = useWallet()
+  const [isEditing, setIsEditing] = useState(false)
+  const handleClickSettings = useCallback(() => {
+    settings.open({})
+    onClose && onClose()
+  }, [onClose, settings])
+
+  const handleClickSupport = useCallback(() => {
+    feedbackSupport.open({})
+    onClose && onClose()
+  }, [onClose, feedbackSupport])
+
+  const handleCreate = useCallback(() => {
+    onClose && onClose()
+    dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
+    create(KeyManager.Mobile)
+  }, [create, dispatch, onClose])
+
+  const handleImport = useCallback(() => {
+    onClose && onClose()
```

Comment:
`onClose` is defined at type-level, can remove checks here and in other event-handlers